draft-marchan-kdl2.md

```diff
@@ -1013,12 +1102,27 @@ multi-line-raw-string-body :=
 // Numbers
 number := keyword-number | hex | octal | binary | decimal
 
-decimal := sign? integer ('.' integer)? exponent?
+decimal := significand exponent?
+suffixed-decimal := significand (
+    bare-type-suffix
+    | (exponent? explicit-type-suffix)
+)
+significand := sign? significand-initial integer? ('.' integer)?
 exponent := ('e' | 'E') sign? integer
 integer := digit (digit | '_')*
+significand-initial = digit
+    - '0b'
+    - '0o'
+    - '0x'
```

Comment:
This use of `-` isn't quite right - it's not subtracting from the LHS (the `digit`), it's imposing a wider restriction on the digit *and* its following character (which in this grammar would come from the start of `bare-type-suffix`).

I think this should instead be handled as a cut point on the `hex`/`octal`/`binary` productions, so that if you see their prefix you're committed. (We'll also have to specify that the ordering of `|` alternates *is* significant when cuts are involved, so `normal-value` is parsed first and gets to claim that syntax space with a cut before `suffixed-decimal` can run and claim it.)